static/js/store/configureStore.js

```diff
@@ -0,0 +1,39 @@
+/* global require:false, module:false */
+import { compose, createStore, applyMiddleware } from 'redux';
+import thunkMiddleware from 'redux-thunk';
+import createLogger from 'redux-logger';
+import rootReducer from '../reducers';
+import { persistState as devToolsPersistState, devTools } from 'redux-devtools';
+
+let createStoreWithMiddleware;
+if (process.env.NODE_ENV !== "production") {
+  createStoreWithMiddleware = compose(
+    applyMiddleware(
+      thunkMiddleware,
+      createLogger()
+    ),
+    devTools(),
+    devToolsPersistState(window.location.href.match(/[?&]debug_session=([^&]+)\b/))
+  )(createStore);
+} else {
+  createStoreWithMiddleware = compose(
+    applyMiddleware(
+      thunkMiddleware
+    )
+  )(createStore);
+}
+
+export default function configureStore(initialState) {
+  const store = createStoreWithMiddleware(rootReducer, initialState);
+
+  if (module.hot) {
```

Comment:
I originally added it to teachersportal, from here: https://webpack.github.io/docs/hot-module-replacement.html